### ISSUE TITLE
Reanimate Imprint 3

### ISSUE THREAD

**Issue by hollerauer** [nextcloud/apps#37]
Original issue (https://github.com/nextcloud/server/issues/5626) was closed in favor of a pull request (https://github.com/nextcloud/apps/pull/36), which was also closed because it was not bug free.

The workaround (use external site) is not meeting the "high standards for quality" (@MariusBluem), which nextcloud is trying to achieve, at all.

Please find a solution how to bring the functionality the imprint app once had back to latest NC release!

This is not a nice to have feature, but a must have for every private or public installation, which can be accessed web based.

**Comment by MariusBluem** [comment]
Please explain, why `external`-app does not solve this for you. You can put external links as an app-icon link, as an link to the menu  or even as quota-link in the personal settings. What do you need more 😮

**Comment by xraMsamohT** [comment]
> Please find a solution how to bring the functionality the imprint app once had back to latest NC release!

@hollerauer This app is _not_ essential for every Nextcloud user, especially not outside of D-A-CH.
Hire someone who updates this 3rd Party app for you when it is a _must have_ for **you**!

**Comment by hollerauer** [comment]
Finally we have a discussion. :-)

@MariusBluem For me one main use of nextcloud for customers is to protect their data. This can be achieved by self-hosting. But what would you think about a product which makes you judicial harmful? I would expect a product that wants to protect my data also protects me personally.
I absolutely agree with you by saying there is a way to deal with it (external sites app). But is it working as expected by users? Do you (as user) want to setup another service or even worth write some html-code myself, just to fulfill judicial requirements?
To provide a ready-to-use box (sw & hw!), speeks a different language!

The Ideally this functionality would be part of the server itself. But an app would be a good starting point.

@xraMsamohT I hope I could make clear, why this is not my personal problem, but a inconsistency in the product we try to develop.

**Comment by xraMsamohT** [comment]
@hollerauer No, this is a regional problem that can be solved with an external app. There's no need to bind server resources for it. I see absolutely no “inconsistency in the product”.

**Comment by hollerauer** [comment]
@xraMsamohT For better understanding we should split this discussion in logical parts:

1. Regional or universal problem: It might be correct that legal info is just necessary in a minor number of countries worldwide. But arguing that this should be the reason why not spending any effort in more usability is hard to accept for me. The same argument leads to the conclusion, that for example an italien translation might be not worth to develop, just because it is spoken just in three countries.

2. My request was not to provide a hardcoded legal info page referring to german law. The idea behind my request is that users should be on the one hands side informed and on the other side guided. And as already explained in my previous post, an external link (and the work that must be done to get there) is not what I as user would expect.

3. Leaving legal info asside, what's about other information that must be published on website? What's about privacy info & terms of service info? I know that some of this is only necessary for commercial installation, but they will even more expect an easy to use & userfriendly solution.

4. The inconsistency I mentioned in a previous post is referring to the fact that nextcloud GmbH or it's partners promote & sell a ready to use product (nextcloud box) to EU and US market without even mentioning that users might have to take care about some legal aspects, when bringing this product (as advertised) online. It's not clear to me if this practice is illegal already, but this must be discussed elsewhere. What I would expect is at least a documented way how to achieve legal conformity from technical side and a recommendation to choose a source for the judicial side.

SUMARY:
To sum this post up I'm still convinced that a separate app with the functionality of imprint is worth the effort spend. This will improve our product!

**Comment by Schmuuu** [comment]
Hi,

I agree with @hollerauer 
External Sites are not a good workaround in my opinion. Overall "External Sites" doesn't work well for me. And as soon as I no longer need it as this unsatisfying workaround for another issue, I will remove it from my server.
It seems pretty obvious that the imprint app provides much more than the external sites. The imprint from the old app was visible on the login screen, for guest (for down- and uploads) and for logged in users - and this is, what a person needs for an imprint. 
From the legal point of view (in Germany) the imprint has to be directly linked and accessible on every site on a web page - and this is exactly what the imprint app offered. I found it pretty awesome.

**Comment by hollerauer** [comment]
bump:
What is the next step?

**Comment by MariusBluem** [comment]
> What is the next step?

Maybe find someone who wants to contribute, or develop it by yourself. You might also consider to post a bountysource bounty on it ;) 
...please dont rely on feature-implementations done by „the Nextclouders“ since we are not able to solve all of your feature requests in under 1 week.

It’s open source - contribute - THX! 😇

**Comment by MichaIng** [comment]
Hey there, sorry for the delay. I want to contribute to the discussion and am going to install and check the app for the issues mentioned. I am absolutely no php programmer, though not too bad in understanding programming/syntax concepts and will give my best to identify and "maybe" starting to solve the problems. First to the discussion:

I agree that nextcloud employed programmers resource is limited and has to concentrate on a stable and secure running product. As the legal part is currently solvable by host/customer contracts and external links, this topic is a minor request that could be well solved community driven or via bountysource. But I also agree that the purpose of this app fits very well to the philosophy on legal and esp. privacy aspects that I have and obviously also others here. Also it fits perfectly to one important reason, why people actually use nextcloud instead of well working other commercial solutions.

>1. Regional or universal problem: It might be correct that legal info is just necessary in a minor number of countries worldwide. But arguing that this should be the reason why not spending any effort in more usability is hard to accept for me. The same argument leads to the conclusion, that for example an italien translation might be not worth to develop, just because it is spoken just in three countries.

Also the legal regulations in stricter counties do have some reasons that I totally agree in many cases. So even that some legal/privacy info might not be necessary in some country, it might fit to ones philosophy in how he/she wants to share his data and/or nextcloud instance with others and how he/she wants to inform other users on that behaviour and how/in what cases possibilities and tracking usage via server will be used or not used.

> 2. My request was not to provide a hardcoded legal info page referring to german law. The idea behind my request is that users should be on the one hands side informed and on the other side guided. And as already explained in my previous post, an external link (and the work that must be done to get there) is not what I as user would expect.

Besides the counties rules, the legal/privacy info one wants to other users is and should be totally individual. It should be not about prescribing hosts how they have to handle their instance/users, more to remind everybody that legal and privacy concerns might be something to think about. Even if I just share my instance to friends and family, it can become a topic how I want others to use my server and how not/which kind of data I want to have stored on my server, and which could bring me into uncomfortable situations. Also I might want to inform others, that/if I have the possibility to track every single step they do, access their data (if not client side encrypted) and if/how I will do use of that possibilities. This is again not just a topic for commercial used nextcloud instances from my point of view.

> 3. Leaving legal info aside, what's about other information that must be published on website? What's about privacy info & terms of service info? I know that some of this is only necessary for commercial installation, but they will even more expect an easy to use & userfriendly solution.

Already said: It is one question, what is necessary, but another how I want to handle the personal rights/trust of my users and how I want to protect myself against perhaps illegal or embarrassing misuse of my nextcloud/server capacity, even if it's just between friends and family.

> 4. The inconsistency I mentioned in a previous post is referring to the fact that nextcloud GmbH or it's partners promote & sell a ready to use product (nextcloud box) to EU and US market without even mentioning that users might have to take care about some legal aspects, when bringing this product (as advertised) online. It's not clear to me if this practice is illegal already, but this must be discussed elsewhere. What I would expect is at least a documented way how to achieve legal conformity from technical side and a recommendation to choose a source for the judicial side.

Everyone, who brings up a commercial web service has to and will inform oneself about the necessary legal parts and will do some kind of contracts with the customers. So from my point of view it might be even more interesting for hosts that do not/net yet go commercial and besides for others, who access ones nextcloud data via shared links etc. To have the wanted legal/privacy etc. info/links on every nextcloud page on the same position visible, is the big strength of this app.

**Comment by 9662** [comment]
Just to understand what is being requested, are we looking at something like the [Libreoffice guys have](https://nextcloud.documentfoundation.org/)? A couple of links like that would be ideal.

Any idea how they implemented that on their ☁️?

**Comment by ShamimIslam** [comment]
Ok guys. I've heard a lot. I love the Imprint App. I'm actually the reason the app was even available past OwnCloud v8.x. I went to the maintainer. As of 2 years ago, he stopped maintaining the app and as his last hurrah, he updated the version information. It used to work in NC 10 and 11. 

Here's the current status of the Imprint App. The original app maintainer has relinquished control of the app. The current NextCloud engine when used with the Imprint app (with a little finagling from me), simply corrupts the menu interface. That's because the way the imprint app worked was based on how the NextCloud interface was set up. I have dropped an email to the originator of the Imprint app.

So - I *CAN* fix this. But I need someone to explain to me how NextCloud allows apps to inject content into the header, into the apps menu and into the body when the app itself is running and how it allows me to provide a template for admin screen data updates and finally how to update the app preferences information.

I can do this but the documentation on the classes and members and methods in OC are few and far between and everyone says read the code. I want to know what parts to read. So rather than read the whole codebaes, if someone could point me to the classes that have the functions I want, I can do the rest.

Thank you.

P.S. I will probably generate the app from scratch so if you want to pick a better name, now is the time.

P.P.S. A customized them and a footer is still a workaround. I don't want this noise on every page. Sheesh. This should be easy. It shouldn't require a designer every time. Come on now. An Administrator should be able to change the legalize to comply with regulations. We should not require a designer to do so. Even in the commercial space, we do not update the footer by hand - the copy is controlled on the back end by CMS which is in turn managed by an administrator under the supervision of the legal team.
